File: sdk/r/R/proxmox.R
#' HiTechCloudProxmox: machine status, power, VMs, IPs, backups, snapshots, bandwidth
#' @param http HttpClient object
#' @return Resource object
#' @export
ProxmoxResource <- function(http) {
  env <- new.env(parent = emptyenv())
  env$http <- http
  class(env) <- "ProxmoxResource"
  env
}

#' GET /api/service/{id}/htcpve/status - Get machine status
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getStatus <- function(self, ...) {
  self$http$get("/api/service/{id}/htcpve/status", list(...))
}

#' POST /api/service/{id}/htcpve/power - Power action (start/stop/shutdown/reboot/reset)
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
powerAction <- function(self, ...) {
  self$http$post("/api/service/{id}/htcpve/power", list(...))
}

#' GET /api/service/{id}/htcpve/vms - List machines
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listVms <- function(self, ...) {
  self$http$get("/api/service/{id}/htcpve/vms", list(...))
}

#' GET /api/service/{id}/htcpve/ips - List addresses
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listIps <- function(self, ...) {
  self$http$get("/api/service/{id}/htcpve/ips", list(...))
}

#' POST /api/service/{id}/htcpve/rdns - Set reverse DNS
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
setRdns <- function(self, ...) {
  self$http$post("/api/service/{id}/htcpve/rdns", list(...))
}

#' GET /api/service/{id}/htcpve/backups - List backups
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listBackups <- function(self, ...) {
  self$http$get("/api/service/{id}/htcpve/backups", list(...))
}

#' POST /api/service/{id}/htcpve/backups - Create backup
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
createBackup <- function(self, ...) {
  self$http$post("/api/service/{id}/htcpve/backups", list(...))
}

#' GET /api/service/{id}/htcpve/snapshots - List snapshots
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listSnapshots <- function(self, ...) {
  self$http$get("/api/service/{id}/htcpve/snapshots", list(...))
}

#' POST /api/service/{id}/htcpve/snapshots - Create snapshot
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
createSnapshot <- function(self, ...) {
  self$http$post("/api/service/{id}/htcpve/snapshots", list(...))
}

#' GET /api/service/{id}/htcpve/usage - Get bandwidth usage
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getBandwidthUsage <- function(self, ...) {
  self$http$get("/api/service/{id}/htcpve/usage", list(...))
}

File: sdk/c/src/proxmox.c
#include "hitechcloud/proxmox.h"
#include <stdlib.h>


/* GET /api/service/{id}/htcpve/status - Get machine status */
char *hc_proxmox_get_status(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/htcpve/status", NULL);
}

/* POST /api/service/{id}/htcpve/power - Power action (start/stop/shutdown/reboot/reset) */
char *hc_proxmox_power_action(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/htcpve/power", NULL);
}

/* GET /api/service/{id}/htcpve/vms - List machines */
char *hc_proxmox_list_vms(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/htcpve/vms", NULL);
}

/* GET /api/service/{id}/htcpve/ips - List addresses */
char *hc_proxmox_list_ips(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/htcpve/ips", NULL);
}

/* POST /api/service/{id}/htcpve/rdns - Set reverse DNS */
char *hc_proxmox_set_rdns(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/htcpve/rdns", NULL);
}

/* GET /api/service/{id}/htcpve/backups - List backups */
char *hc_proxmox_list_backups(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/htcpve/backups", NULL);
}

/* POST /api/service/{id}/htcpve/backups - Create backup */
char *hc_proxmox_create_backup(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/htcpve/backups", NULL);
}

/* GET /api/service/{id}/htcpve/snapshots - List snapshots */
char *hc_proxmox_list_snapshots(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/htcpve/snapshots", NULL);
}

/* POST /api/service/{id}/htcpve/snapshots - Create snapshot */
char *hc_proxmox_create_snapshot(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/htcpve/snapshots", NULL);
}

/* GET /api/service/{id}/htcpve/usage - Get bandwidth usage */
char *hc_proxmox_get_bandwidth_usage(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/htcpve/usage", NULL);
}

